src/ServicePulse.Host/app/js/views/pending_retries/controller.js

```diff
@@ -0,0 +1,375 @@
+﻿/// <reference path="service.js" />
+;
+(function(window, angular, undefined) {
+    "use strict";
+
+    function controller(
+        $scope,
+        $timeout,
+        $location,
+        $moment,
+        $filter,
+        scConfig,
+        toastService,
+        sharedDataService,
+        notifyService,
+        serviceControlService,
+        endpointsService, 
+        redirectService,
+        pendingRetryService) {
+
+        var vm = this;
+
+        var notifier = notifyService();
+        
+        vm.sortButtonText = '';
+        vm.sortDirection = 'asc';
+        vm.allMessagesLoaded = false;
+        vm.loadingData = false;
+
+        vm.timeGroup = {
+            amount: undefined,
+            unit: undefined,
+            buttonText: 'All Pending Retries',
+            selected: function () {
+                return $moment.duration(vm.timeGroup.amount, vm.timeGroup.unit);;
+            }
+        };
+
+        notifier.subscribe($scope, function (event, data) {
+            if (vm.total > data) {
+                removeResolvedMessages();
+            }
+
+            if (vm.total !== data) {
+                vm.total = data;
+                vm.loadTotalBasedOnFilters();
+                vm.loadMoreResults();
+            } 
+        }, 'PendingRetriesTotalUpdated');
+
+        notifier.subscribe($scope, function (event, data) {
+            vm.endpoints = data;
+        }, 'EndpointsUpdated');
+
+        notifier.subscribe($scope, function (event, response) {
+            vm.redirects = response.data;
+            refreshRedirects();
+        }, 'RedirectsUpdated');
+
+        var setSortButtonText = function (sort, direction) {
+            vm.sortButtonText = (sort === 'message_type' ? "Message Type" : "Time of Failure");
+            vm.sortDirection = direction;
+        }
+
+        function removeResolvedMessages() {
+            vm.pendingRetryMessages = vm.pendingRetryMessages.filter(function (item) {
+                return !item.resolved;
+            });
+        }
+
+        function refreshRedirects() {
+            vm.pendingRetryMessages.map(function(obj) {
```

Comment:
I don't think this is doing anything. `map` returns the new list, but we don't save the return result anywhere.

If we're simply iterating over the list to modify it and the result doesn't matter, I feel it would be better to use `forEach` to better represent that intention.